### PR TITLE
feat: add design system button variants

### LINKED ISSUE #30: [DEV] Buttons
Finalize the button development against the design. Also for a POC we can try incorporating props to use variant & size for easier development and less copy paste issues.

### PR DESCRIPTION
**Pull request by Matgorn**
PR includes:
- chore: add color palette to tailwind config
- feat: add button variants based on [figma](https://www.figma.com/file/aAZNLti1x7RHcKO2aaVdyh/Selleo-Design-System?node-id=1179%3A338&t=2Uhcfa8c1Jgjj4I2-4)

closes #30

### CODE REVIEW

File: packages/selleo-design-core/src/Button.tsx [hunk 1]
```diff
@@ -1,9 +1,25 @@
 import { h } from "preact";
 
-export function Button() {
-  return <button class="inline-flex bg-orange-500">Button</button>;
+export function PrimaryButtonSmall() {
+  return <button class="inline-flex justify-center text-sm text-white font-extrabold leading-7 tracking-normal min-w-[124px] min-h-[32px] bg-brand-primary hover:bg-brand-primary-600 focus:bg-brand-primary-600 focus:outline focus:outline-offset-1 focus:outline-1 focus:outline-brand-primary-600 rounded disabled:bg-brand-primary-500 disabled:opacity-30">Text</button>;
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:4]: can you check how will it behave when we change from focus: to focus-visible to get rid of focus state when clicking on it?

Comment (k1eu) [reply packages/selleo-design-core/src/Button.tsx:4]: @michalgren what do you think?

File: packages/selleo-tailwind/tailwind.config.cjs [hunk 2]
```diff
@@ -14,10 +14,41 @@ module.exports = {
...
+    letterSpacing: {
+      normal: '0.01em',
+    },
     extend: {
       colors: {
-				'primary-500': '#FAFAFA',
-			}
+        brand: {
+          primary: {
+            DEFAULT: '#ff6d2a',
```
Comment (k1eu) [inline packages/selleo-tailwind/tailwind.config.cjs:36]: i think i'd get rid of DEFAULT key in the colors
 we can leave a comment next to 500 key `// default`

**Review by k1eu — CHANGES_REQUESTED** [review]

**Comment by k1eu** [comment]
I added @michalgren as a Designer Reviewer

File: packages/selleo-design-core/src/Button.tsx [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
+  return <button class="inline-flex justify-center text-base text-white font-extrabold leading-7 tracking-normal min-w-[160px] min-h-[48px] bg-brand-primary hover:bg-brand-primary-600 focus:bg-brand-primary-600 focus:outline focus:outline-1 focus:outline-brand-primary-600 focus:outline-offset-2 rounded disabled:bg-brand-primary-500 disabled:opacity-30">Text</button>;
+}
+
+export function PrimaryOutlinedButtonSmall() {
+  return <button class="inline-flex justify-center text-sm font-extrabold min-w-[124px] min-h-[32px] text-brand-primary border border-solid border-brand-primary bg-white hover:bg-brand-primary-100 focus:bg-brand-primary-100 focus:outline focus:outline-1 focus:outline-offset-2 focus:outline-brand-primary rounded disabled:bg-white disabled:opacity-30">Text</button>;
 }
 
-export function ButtonExample() {
-  return <button class="inline-flex bg-orange-700">Button</button>;
+export function PrimaryOutlinedButton() {
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:15]: I guess the Background should be clear instead of white @michalgren am i right?

**Review by k1eu — CHANGES_REQUESTED** [review]

File: packages/selleo-design-core/src/Button.tsx (hunk 1 above)
Comment (Matgorn) [reply packages/selleo-design-core/src/Button.tsx]: Changes to `:focus-visible` so it's accessible via `tab` but not visible after button click.

File: packages/selleo-tailwind/tailwind.config.cjs (hunk 2 above)
Comment (Matgorn) [reply packages/selleo-tailwind/tailwind.config.cjs]: Done

File: packages/selleo-design-core/src/Button.tsx (hunk 3 above)
Comment (Matgorn) [reply packages/selleo-design-core/src/Button.tsx:15]: IMO It makes more sense like this indeed, for now changed according to @k1eu comments 👍

**Comment by michalgren** [comment]
Changes:
Button-sm - without min-width
      font-weight: 700
      font-size: 14px
      padding: 8px 16px
      gap: 4px

Button-base - without min-width
      font-weight: 700
      font-size: 16px
      padding: 8px 24px
      gap: 8px

We could add some animation to switch between states. 300ms e.g.
Adjust link styles - I show them on the [designs](https://www.figma.com/file/aAZNLti1x7RHcKO2aaVdyh/Selleo-Design-System?node-id=2354%3A7625&t=nvC9SvC8aNaBsptj-11)

File: packages/selleo-design-core/src/Button.tsx [hunk 4]
```diff
@@ -1,61 +1,29 @@
 import { h } from "preact";
+import classNames from "classnames";
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:2]: `import cx from "classnames"

with cx will be easier to write

File: packages/selleo-design-core/src/Button.tsx [hunk 5]
```diff
@@ -1,61 +1,29 @@
...
 
-export function PrimaryButtonSmall() {
-  return <button class="inline-flex py-1 px-2 gap-0.5 transition duration-300 justify-center text-xs text-white font-bold leading-7 tracking-normal bg-brand-primary-500 hover:bg-brand-primary-600 focus-visible:bg-brand-primary-600 focus-visible:outline focus-visible:outline-offset-1 focus-visible:outline-1 focus-visible:outline-brand-primary-600 rounded disabled:bg-brand-primary-500 disabled:opacity-30">Text</button>;
-}
-
-export function PrimaryButton() {
-  return <button class="inline-flex py-1 px-3 gap-1 transition duration-300 justify-center text-sm text-white font-bold leading-7 tracking-normal bg-brand-primary-500 hover:bg-brand-primary-600 focus-visible:bg-brand-primary-600 focus-visible:outline focus-visible:outline-1 focus-visible:outline-brand-primary-600 focus-visible:outline-offset-2 rounded disabled:bg-brand-primary-500 disabled:opacity-30">Text</button>;
-}
+export function Button(
+  { primaryOutlined, link, small, ...rest }: { primaryOutlined?: boolean, link?: boolean, small?: boolean } & h.JSX.HTMLAttributes<HTMLButtonElement>
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:5]: let's have props like `variant = 'primary'  | 'primaryOutlined' etc`
and `size='small' | 'normal'` with default values being primary and normal

**Review by k1eu — CHANGES_REQUESTED** [review]

**Review by k1eu — APPROVED** [review]

**Pull request merged**